diff --git a/src/applications/maniphest/editor/ManiphestTransactionEditor.php b/src/applications/maniphest/editor/ManiphestTransactionEditor.php
index 0a4ea84..0339041 100644
--- a/src/applications/maniphest/editor/ManiphestTransactionEditor.php
+++ b/src/applications/maniphest/editor/ManiphestTransactionEditor.php
@@ -219,6 +219,7 @@ final class ManiphestTransactionEditor
 
         id(new PhabricatorEdgeEditor())
           ->addEdge($parent_phid, $parent_type, $task_phid)
+	  ->setDangerousOverrideTimeStamp($xaction->getDangerousOverrideTimeStamp())
           ->save();
         break;
       case PhabricatorTransactions::TYPE_COLUMNS:
@@ -266,6 +267,7 @@ final class ManiphestTransactionEditor
         foreach ($blocked_tasks as $blocked_task) {
           $parent_xaction = id(new ManiphestTransaction())
             ->setTransactionType(ManiphestTransaction::TYPE_UNBLOCK)
+	    ->setDangerousOverrideTimeStamp($unblock_xaction->getDangerousOverrideTimeStamp())
             ->setOldValue(array($object->getPHID() => $old))
             ->setNewValue(array($object->getPHID() => $new));
 
@@ -894,6 +896,7 @@ final class ManiphestTransactionEditor
         if ($old_value) {
           $results[] = id(new ManiphestTransaction())
             ->setTransactionType(PhabricatorTransactions::TYPE_SUBSCRIBERS)
+	    ->setDangerousOverrideTimeStamp($xaction->getDangerousOverrideTimeStamp())
             ->setIgnoreOnNoEffect(true)
             ->setNewValue(
               array(
diff --git a/src/applications/transactions/editor/PhabricatorApplicationTransactionEditor.php b/src/applications/transactions/editor/PhabricatorApplicationTransactionEditor.php
index 5851e01..c493d5e 100644
--- a/src/applications/transactions/editor/PhabricatorApplicationTransactionEditor.php
+++ b/src/applications/transactions/editor/PhabricatorApplicationTransactionEditor.php
@@ -1450,6 +1450,7 @@ abstract class PhabricatorApplicationTransactionEditor
     }
 
     $xaction = newv(get_class(head($xactions)), array());
+    $xaction->setDangerousOverrideTimeStamp(head($xactions)->getDangerousOverrideTimeStamp());
     $xaction->setTransactionType(PhabricatorTransactions::TYPE_SUBSCRIBERS);
     $xaction->setNewValue(array('+' => $phids));
 
@@ -2394,9 +2395,11 @@ abstract class PhabricatorApplicationTransactionEditor
     }
 
     $should_cc = false;
+    $override_ts = null;
     foreach ($xactions as $xaction) {
       if ($this->shouldImplyCC($object, $xaction)) {
         $should_cc = true;
+	$override_ts = $xaction->getDangerousOverrideTimeStamp();
         break;
       }
     }
@@ -2426,7 +2429,9 @@ abstract class PhabricatorApplicationTransactionEditor
     $xaction = newv(get_class(head($xactions)), array());
     $xaction->setTransactionType(PhabricatorTransactions::TYPE_SUBSCRIBERS);
     $xaction->setNewValue(array('+' => array($actor_phid)));
-
+    if ($override_ts !== null) {
+      $xaction->setDangerousOverrideTimeStamp($override_ts);
+    }
     array_unshift($xactions, $xaction);
 
     return $xactions;
@@ -3045,10 +3050,11 @@ abstract class PhabricatorApplicationTransactionEditor
     $story_type = $this->getFeedStoryType();
     $story_data = $this->getFeedStoryData($object, $xactions);
 
+    $xaction = $xactions[array_keys($xactions)[0]];
     id(new PhabricatorFeedStoryPublisher())
       ->setStoryType($story_type)
       ->setStoryData($story_data)
-      ->setStoryTime(time())
+      ->setStoryTime($xaction->getDateCreated())
       ->setStoryAuthorPHID($this->getActingAsPHID())
       ->setRelatedPHIDs($related_phids)
       ->setPrimaryObjectPHID($object->getPHID())
@@ -3317,6 +3323,7 @@ abstract class PhabricatorApplicationTransactionEditor
       $template
         ->setTransactionType($xaction->getTransactionType())
         ->setMetadataValue('edge:type', $inverse_type)
+	->setDangerousOverrideTimeStamp($xaction->getDangerousOverrideTimeStamp())
         ->setNewValue(
           array(
             $edge_edit_type => array($object->getPHID() => $object->getPHID()),
diff --git a/src/infrastructure/storage/lisk/LiskDAO.php b/src/infrastructure/storage/lisk/LiskDAO.php
index 0527c3c..04dd6c8 100644
--- a/src/infrastructure/storage/lisk/LiskDAO.php
+++ b/src/infrastructure/storage/lisk/LiskDAO.php
@@ -199,6 +199,9 @@ abstract class LiskDAO extends Phobject {
   protected $dateCreated;
   protected $dateModified;
 
+  private $dangerousOverrideID;
+  private $dangerousOverrideTimeStamp;
+
   /**
    *  Build an empty object.
    *
@@ -212,6 +215,38 @@ abstract class LiskDAO extends Phobject {
   }
 
 
+/* -(  Shooting Yourself in the Foot   )------------------------------------- */
+
+  /*
+   * Both of these "dangerous" methods are for overriding when objects
+   * are created/changed and what their IDs are.  This is intended for
+   * (A) writing import scripts from other systems (B) setting up test
+   * scenarios.
+   *
+   * They are dangerous and there is little safety to keep you from
+   * wrecking an install.  There is no guarantee that future versions
+   * of Phabricator will not place logical constraints on the ordering
+   * of timestamps and bad things may happen if you create impossible scenarios.
+   */
+  public function setDangerousOverrideID($id) {
+    $this->dangerousOverrideID = $id;
+    return $this;
+  }
+
+  public function setDangerousOverrideTimeStamp($ts) {
+    $this->dangerousOverrideTimeStamp = $ts;
+    return $this;
+  }
+
+  public function getDangerousOverrideTimeStamp() {
+    if (isset($this->dangerousOverrideTimeStamp)) {
+      return $this->dangerousOverrideTimeStamp;
+    } else {
+      return null;
+    }
+  }
+
+
 /* -(  Managing Connections  )----------------------------------------------- */
 
 
@@ -1200,7 +1235,6 @@ abstract class LiskDAO extends Phobject {
     $data = $this->getAllLiskPropertyValues();
 
     $conn = $this->establishConnection('w');
-
     $id_mechanism = $this->getConfigOption(self::CONFIG_IDS);
     switch ($id_mechanism) {
       case self::IDS_AUTOINCREMENT:
@@ -1208,7 +1242,10 @@ abstract class LiskDAO extends Phobject {
         // ID column, if it is empty. If the caller has explicitly provided a
         // value, use it.
         $id_key = $this->getIDKeyForUse();
-        if (empty($data[$id_key])) {
+        if (isset($this->dangerousOverrideID)) {
+          $this->setID($this->dangerousOverrideID);
+          $data[$id_key] = $this->dangerousOverrideID;
+        } else if (empty($data[$id_key])) {
           unset($data[$id_key]);
         }
         break;
@@ -1216,7 +1253,10 @@ abstract class LiskDAO extends Phobject {
         // If we are using counter IDs, assign a new ID if we don't already have
         // one.
         $id_key = $this->getIDKeyForUse();
-        if (empty($data[$id_key])) {
+        if (isset($this->dangerousOverrideID)) {
+          $this->setID($this->dangerousOverrideID);
+          $data[$id_key] = $this->dangerousOverrideID;
+        } else if (empty($data[$id_key])) {
           $counter_name = $this->getTableName();
           $id = self::loadNextCounterValue($conn, $counter_name);
           $this->setID($id);
@@ -1377,10 +1417,15 @@ abstract class LiskDAO extends Phobject {
     $use_timestamps = $this->getConfigOption(self::CONFIG_TIMESTAMPS);
 
     if ($use_timestamps) {
-      if (!$this->getDateCreated()) {
-        $this->setDateCreated(time());
+      if (isset($this->dangerousOverrideTimeStamp)) {
+        $this->setDateCreated($this->dangerousOverrideTimeStamp);
+        $this->setDateModified($this->dangerousOverrideTimeStamp);
+      } else {
+        if (!$this->getDateCreated()) {
+          $this->setDateCreated(time());
+        }
+        $this->setDateModified(time());
       }
-      $this->setDateModified(time());
     }
 
     if ($this->getConfigOption(self::CONFIG_AUX_PHID) && !$this->getPHID()) {
